tests/kafkatest/tests/streams/streams_broker_compatibility_test.py

```diff
@@ -66,14 +67,11 @@ def test_compatible_brokers(self, broker_version):
 
         self.processor.wait()
 
-        num_consumed_mgs = self.consumer.total_consumed()
+        wait_until(lambda: self.consumer.total_consumed() == 1, timeout_sec=30, err_msg="Did expect to read exactly one message but got none within 30 seconds")
```

Comment:
Right, I was referring to the difference to the old assert and that the new check and error message contains less troubleshooting information than the old one, but I understand that it might seem irrelevant given that the producer part can be trusted, thus the "nit" :)